### ISSUE TITLE
Node JS 16 error

### ISSUE THREAD

**Issue by VishalSingh28** [eclipse-theia/theia#10677]
1. Unable to run the program on node js versioon 16

**Comment by vince-fugnitto** [comment]
@VishalSingh28 thank you for reporting the issue, in the future please be sure to properly fill out the issue template.
What exactly fails for you when building with `node 16`? Note that you will need to use `--ignore-engines` when building as described in the [prerequisites](https://github.com/eclipse-theia/theia/blob/master/doc/Developing.md#prerequisites)

**Comment by VishalSingh28** [comment]
Actually I have created theia web ide according to your given steps but In your document it saying node required node version is 12 also when I try to run the code with version 16 getting the error and not able to run the code

**Comment by vince-fugnitto** [comment]
>...when I try to run the code with version 16 getting the error and not able to run the code

@VishalSingh28 what is the error?

**Comment by VishalSingh28** [comment]
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'dugite-extra@0.1.14',
npm WARN EBADENGINE   required: { node: '>=10.11.0 <13' },
npm WARN EBADENGINE   current: { node: 'v16.13.2', npm: '8.1.2' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'find-git-exec@0.0.3',
npm WARN EBADENGINE   required: { node: '>=10.11.0 <13' },
npm WARN EBADENGINE   current: { node: 'v16.13.2', npm: '8.1.2' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'back@0.1.5',
npm WARN EBADENGINE   required: { node: '0.10.x || 0.8.x' },
npm WARN EBADENGINE   current: { node: 'v16.13.2', npm: '8.1.2' }
npm WARN EBADENGINE }
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated vscode-debugprotocol@1.51.0: This package has been renamed to @vscode/debugprotocol, please update to the new name
npm WARN deprecated cross-spawn-async@2.2.5: cross-spawn no longer requires a build toolchain, use it instead

> prepare
> yarn run clean && yarn build && yarn run download:plugins

yarn run v1.22.17
$ theia clean
Done in 0.68s.
yarn run v1.22.17
$ theia build --mode development
Failed to resolve module: filenamify
Error: spawn /home/arpit-iot/theiaa/node_modules/@theia/application-manager/node_modules/.bin/webpack ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:282:19)
    at onErrorNT (node:internal/child_process:477:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawn /home/arpit-iot/theiaa/node_modules/@theia/application-manager/node_modules/.bin/webpack',
  path: '/home/arpit-iot/theiaa/node_modules/@theia/application-manager/node_modules/.bin/webpack',
  spawnargs: [ '--mode', 'development' ]
}

**Comment by VishalSingh28** [comment]
yarn install v1.22.17
info No lockfile found.
warning package-lock.json found. Your project contains lock files generated by tools other than Yarn. It is advised not to mix package managers in order to avoid resolution inconsistencies caused by unsynchronized lock files. To clear this warning, remove package-lock.json.
[1/4] Resolving packages...
warning @theia/callhierarchy > @theia/core > @theia/application-package > request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
warning @theia/callhierarchy > @theia/core > @theia/application-package > request > uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
warning @theia/callhierarchy > @theia/core > @theia/application-package > request > har-validator@5.1.5: this library is no longer supported
warning @theia/callhierarchy > @theia/monaco > @theia/filesystem > trash > uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
warning @theia/callhierarchy > @theia/monaco > @theia/filesystem > trash > xdg-trashdir > @sindresorhus/df > execa > cross-spawn-async@2.2.5: cross-spawn no longer requires a build toolchain, use it instead
warning @theia/file-search > vscode-ripgrep@1.13.2: This package has been renamed to @vscode/ripgrep, please update to the new name
warning @theia/git > dugite-extra > dugite-no-gpl > request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
warning @theia/plugin-ext-vscode > request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
warning @theia/plugin-ext-vscode > @theia/plugin-ext > request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
warning @theia/plugin-ext-vscode > @theia/plugin-ext > vscode-debugprotocol@1.51.0: This package has been renamed to @vscode/debugprotocol, please update to the new name
warning @theia/plugin-ext-vscode > @theia/plugin-ext > @theia/search-in-workspace > vscode-ripgrep@1.13.2: This package has been renamed to @vscode/ripgrep, please update to the new name
warning @theia/plugin-ext-vscode > @theia/plugin-ext > @theia/debug > vscode-debugprotocol@1.51.0: This package has been renamed to @vscode/debugprotocol, please update to the new name
warning @theia/preview > @types/highlight.js@10.1.0: This is a stub types definition. highlight.js provides its own type definitions, so you do not need this installed.
warning @theia/search-in-workspace > vscode-ripgrep@1.13.2: This package has been renamed to @vscode/ripgrep, please update to the new name
warning @theia/cli > mocha > debug@3.2.6: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
warning @theia/cli > mocha > chokidar > fsevents@2.1.3: "Please update to latest v2.3 or v2.2"
warning @theia/cli > @theia/application-manager > electron-rebuild > node-gyp > request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
[2/4] Fetching packages...
error dugite-extra@0.1.14: The engine "node" is incompatible with this module. Expected version ">=10.11.0 <13". Got "16.13.2"
error Found incompatible module

**Comment by vince-fugnitto** [comment]
@VishalSingh28 are you still having issues with the build following https://github.com/eclipse-theia/theia/issues/10677#issuecomment-1023523449?

**Comment by VishalSingh28** [comment]
Yes I am getting the error and also want to know how to make live it on Apache server

**Comment by vince-fugnitto** [comment]
> Yes I am getting the error and also want to know how to make live it on Apache server

@VishalSingh28 if you are still getting the error can you please provide the step by step commands you executed?

**Comment by VishalSingh28** [comment]
Actually I want to know about process that I have confusion
